pkg/resolution/resolver/cluster/resolver_test.go

```diff
@@ -0,0 +1,305 @@
+				t.Fatalf("got no error, but expected: %s", tc.expectedErr)
+			}
+			if d := cmp.Diff(tc.expectedErr, err.Error()); d != "" {
+				t.Errorf("error did not match: %s", diff.PrintWantGot(d))
+			}
+		})
+	}
+}
+
+func TestResolve(t *testing.T) {
+	exampleTask := &v1beta1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "example-task",
+			Namespace:       "task-ns",
+			ResourceVersion: "00002",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       string(v1beta1.NamespacedTaskKind),
+			APIVersion: "tekton.dev/v1beta1",
+		},
+		Spec: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Name:    "some-step",
+				Image:   "some-image",
+				Command: []string{"something"},
+			}},
+		},
+	}
+	taskAsYAML, err := yaml.Marshal(exampleTask)
+	if err != nil {
+		t.Fatalf("couldn't marshal task: %v", err)
+	}
+
+	examplePipeline := &v1beta1.Pipeline{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "example-pipeline",
+			Namespace:       "pipeline-ns",
+			ResourceVersion: "00001",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pipeline",
+			APIVersion: "tekton.dev/v1beta1",
+		},
+		Spec: v1beta1.PipelineSpec{
+			Tasks: []v1beta1.PipelineTask{{
+				Name: "some-pipeline-task",
+				TaskRef: &v1beta1.TaskRef{
+					Name: "some-task",
+					Kind: v1beta1.NamespacedTaskKind,
+				},
+			}},
+		},
+	}
+	pipelineAsYAML, err := yaml.Marshal(examplePipeline)
+	if err != nil {
+		t.Fatalf("couldn't marshal pipeline: %v", err)
+	}
+
+	testCases := []struct {
+		name           string
+		kind           string
+		resourceName   string
+		namespace      string
+		expectedStatus *v1alpha1.ResolutionRequestStatus
+		expectedErr    error
+	}{
+		{
+			name:         "successful task",
+			kind:         "task",
+			resourceName: exampleTask.Name,
+			namespace:    exampleTask.Namespace,
+			expectedStatus: &v1alpha1.ResolutionRequestStatus{
+				Status: duckv1.Status{},
+				ResolutionRequestStatusFields: v1alpha1.ResolutionRequestStatusFields{
+					Data: base64.StdEncoding.Strict().EncodeToString(taskAsYAML),
+				},
+			},
+		}, {
+			name:         "successful pipeline",
+			kind:         "pipeline",
+			resourceName: examplePipeline.Name,
+			namespace:    examplePipeline.Namespace,
+			expectedStatus: &v1alpha1.ResolutionRequestStatus{
+				Status: duckv1.Status{},
+				ResolutionRequestStatusFields: v1alpha1.ResolutionRequestStatusFields{
+					Data: base64.StdEncoding.Strict().EncodeToString(pipelineAsYAML),
+				},
+			},
+		}, {
+			name:         "no such task",
+			kind:         "task",
+			resourceName: exampleTask.Name,
+			namespace:    "other-ns",
+			expectedStatus: &v1alpha1.ResolutionRequestStatus{
+				Status: duckv1.Status{
+					Conditions: duckv1.Conditions{{
+						Type:   apis.ConditionSucceeded,
+						Status: corev1.ConditionFalse,
+						Reason: resolutioncommon.ReasonResolutionFailed,
+					}},
+				},
+			},
+			expectedErr: errors.New(`error getting "Cluster" "foo/rr": tasks.tekton.dev "example-task" not found`),
```

Comment:
what are "cluster" and "foo/rr" here? can this error message be clarified?

Reply:
For now, I'm updating the test (and the `git/resolver_test.go` tests, one of which was a duplicate from before the `branch` or `commit` -> `revision` parameter change) to use `&resolutioncommon.ErrorGettingResource{...}` rather than `errors.New(...)` to at least make the test cases a bit cleaner.

Reply:
And created #5409 for this.

Reply:
thanks!